File: Examen_Unidad2/src/Animal.java
import java.util.*;

class Animal {
    private String id;
    private String tipoAnimal;
    private String fechaNacimiento;
    private String fechaLlegada;
    private double peso;
    private List<String> enfermedades;
    private String frecuenciaAlimentacion;
    private String tipoAlimentacion;
    private boolean vacunas;

    public Animal(String tipoAnimal, String fechaNacimiento, String fechaLlegada, String llegada, double peso, List<String> enfermedades,
                  String frecuenciaAlimentacion, String tipoAlimentacion, boolean vacunas) {
        this.id = generarIdAleatorio();
        this.tipoAnimal = tipoAnimal;
        this.fechaNacimiento = fechaNacimiento;
        this.fechaLlegada = fechaLlegada;
        this.peso = peso;
        this.enfermedades = enfermedades;
        this.frecuenciaAlimentacion = frecuenciaAlimentacion;
        this.tipoAlimentacion = tipoAlimentacion;
        this.vacunas = vacunas;
    }

    private String generarIdAleatorio() {
        Random random = new Random();
        int num = random.nextInt(1000) + 1;
        return "A" + num;
    }

    public String getId() {
        return id;
    }

    public String getTipoAnimal() {
        return tipoAnimal;
    }

    public void setTipoAnimal(String tipoAnimal) {
        this.tipoAnimal = tipoAnimal;
    }

    public String getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(String fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public String getFechaLlegada() {
        return fechaLlegada;
    }

    public void setFechaLlegada(String fechaLlegada) {
        this.fechaLlegada = fechaLlegada;
    }

    public double getPeso() {
        return peso;
    }

    public void setPeso(double peso) {
        this.peso = peso;
    }

    public List<String> getEnfermedades() {
        return enfermedades;
    }

    public void setEnfermedades(List<String> enfermedades) {
        this.enfermedades = enfermedades;
    }

    public String getFrecuenciaAlimentacion() {
        return frecuenciaAlimentacion;
    }

    public void setFrecuenciaAlimentacion(String frecuenciaAlimentacion) {
        this.frecuenciaAlimentacion = frecuenciaAlimentacion;
    }

    public String getTipoAlimentacion() {
        return tipoAlimentacion;
    }

    public void setTipoAlimentacion(String tipoAlimentacion) {
        this.tipoAlimentacion = tipoAlimentacion;
    }

    public boolean isVacunas() {
        return vacunas;
    }

    public void setVacunas(boolean vacunas) {
        this.vacunas = vacunas;
    }

    @Override
    public String toString() {
        return "Animal" +
                "Id: '" + id + '\'' +
                ", Tipo Animal: '" + tipoAnimal + '\'' +
                ", Fecha Nacimiento: '" + fechaNacimiento + '\'' +
                ", Fecha Llegada: '" + fechaLlegada + '\'' +
                ", Peso: " + peso +
                ", Enfermedades: " + enfermedades +
                ", Frecuencia Alimentacion: ' " + frecuenciaAlimentacion + '\'' +
                ", Tipo Alimentacion:'" + tipoAlimentacion + '\'' +
                ", Vacunas:" + vacunas;
    }
}

File: Examen_Unidad2 py/ZOO PY/Animal.py
import random
    #clase
class Animal:
        #crear instancia
        #metodo con parametros
        def __init__(self, tipoAnimal, fechaNacimiento, fechaLlegada, peso, enfermedades, frecuenciaAlimentacion, tipoAlimentacion, vacunas):
            self.id = self.generarIdAleatorio()
            self.tipoAnimal = tipoAnimal
            self.fechaNacimiento = fechaNacimiento      #asignar 
            self.fechaLlegada = fechaLlegada
            self.peso = peso
            self.enfermedades = enfermedades
            self.frecuenciaAlimentacion = frecuenciaAlimentacion
            self.tipoAlimentacion = tipoAlimentacion
            self.vacunas = vacunas
        #metodos
        def generarIdAleatorio(self):
            num = random.randint(1, 1000)
            return "A" + str(num)
        #metodos
        def getId(self):
            return self.id
        #metodos
        def getTipoAnimal(self):
            return self.tipoAnimal
        #metodos
        def setTipoAnimal(self, tipoAnimal):
            self.tipoAnimal = tipoAnimal
        #metodos
        def getFechaNacimiento(self):
            return self.fechaNacimiento
        #metodos
        def setFechaNacimiento(self, fechaNacimiento):
            self.fechaNacimiento = fechaNacimiento
        #metodos
        def getFechaLlegada(self):
            return self.fechaLlegada
        #metodos
        def setFechaLlegada(self, fechaLlegada):
            self.fechaLlegada = fechaLlegada
        #metodos
        def getPeso(self):
            return self.peso
        #metodos
        def setPeso(self, peso):
            self.peso = peso
        #metodos
        def getEnfermedades(self):
            return self.enfermedades
        #metodos
        def setEnfermedades(self, enfermedades):
            self.enfermedades = enfermedades
        #metodos
        def getFrecuenciaAlimentacion(self):
            return self.frecuenciaAlimentacion

        def setFrecuenciaAlimentacion(self, frecuenciaAlimentacion):
            self.frecuenciaAlimentacion = frecuenciaAlimentacion

        def getTipoAlimentacion(self):
            return self.tipoAlimentacion

        def setTipoAlimentacion(self, tipoAlimentacion):
            self.tipoAlimentacion = tipoAlimentacion

        def isVacunas(self):
            return self.vacunas

        def setVacunas(self, vacunas):
            self.vacunas = vacunas

        def __str__(self):
            return f"Animal: Id: '{self.id}', Tipo Animal: '{self.tipoAnimal}', Fecha Nacimiento: '{self.fechaNacimiento}', Fecha Llegada: '{self.fechaLlegada}', 
            Peso: {self.peso}, Enfermedades: {self.enfermedades}, Frecuencia Alimentacion: '{self.frecuenciaAlimentacion}', Tipo Alimentacion: '{self.tipoAlimentacion}', 
            Vacunas: {self.vacunas}"

    # Ejemplo de uso
    # animal1 = Animal("León", "01/01/2010", "01/05/2015", 150, ["Gripe"], "2 veces al día", "Carnívoro", True)
    # print(animal1)
